agent/textwatcher.go

```diff
@@ -0,0 +1,194 @@
+package agent
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"hash/crc32"
+)
+
+type TextWatcher interface {
+	Start()
+	Stop()
+	Contents() ([]byte, error)
+	Path() string
+	Kind() string
+	MarshalJSON() ([]byte, error)
+}
+
+type textWatcher struct {
+	sync.Mutex
+	keepPolling  bool
+	kind         string
+	path         string
+	UpdatedAt    time.Time
+	BeingWatched bool
+	OnChange     ChangeHandler
+	Checksum     uint32
+	CmdName      string
+	CmdArgs      []string
+	contents     func() ([]byte, error)
+	pollSleep    time.Duration
+}
+
+// File watchers track changes in the contents of a file
+func NewFileWatcher(path string, callback ChangeHandler) Watcher {
+	var kind string
+	filename := filepath.Base(path)
+	switch filename {
```

Comment:
first part: the kind attribute we receive in the payload is kind of useless. If the agent just looks at the file name in order to set the kind, we can do that just fine in the server also.

the second part: we can do it as a separate pr.